addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/handler/SatelSystemHandler.java

```diff
@@ -0,0 +1,138 @@
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (bridgeHandler != null && CHANNEL_USER_CODE.equals(channelUID.getId()) && command instanceof StringType) {
+            bridgeHandler.setUserCode(command.toFullString());
+        } else {
+            super.handleCommand(channelUID, command);
+        }
+    }
+
+    @Override
+    public void incomingEvent(SatelEvent event) {
+        if (event instanceof IntegraStatusEvent) {
+            logger.trace("Handling incoming event: {}", event);
+            IntegraStatusEvent statusEvent = (IntegraStatusEvent) event;
+            if (thingConfig.isCommandOnly()) {
+                return;
+            }
+            updateState(CHANNEL_DATE_TIME, new DateTimeType(statusEvent.getIntegraTime()));
+            updateSwitch(CHANNEL_SERVICE_MODE, statusEvent.inServiceMode());
+            updateSwitch(CHANNEL_TROUBLES, statusEvent.troublesPresent());
+            updateSwitch(CHANNEL_TROUBLES_MEMORY, statusEvent.troublesMemory());
+            updateSwitch(CHANNEL_ACU100_PRESENT, statusEvent.isAcu100Present());
+            updateSwitch(CHANNEL_INTRX_PRESENT, statusEvent.isIntRxPresent());
+            updateSwitch(CHANNEL_GRADE23_SET, statusEvent.isGrade23Set());
+        } else {
+            super.incomingEvent(event);
+        }
+    }
+
+    @Override
+    protected StateType getStateType(String channelId) {
+        return null;
+    }
+
+    @Override
+    protected SatelCommand convertCommand(ChannelUID channel, Command command) {
+        switch (channel.getId()) {
+            case CHANNEL_TROUBLES:
+            case CHANNEL_TROUBLES_MEMORY:
+                if (command == OnOffType.ON) {
+                    return null;
+                } else {
+                    return new ClearTroublesCommand(bridgeHandler.getUserCode());
+                }
+            case CHANNEL_DATE_TIME:
+                DateTimeType dateTime = null;
+                if (command instanceof StringType) {
+                    dateTime = DateTimeType.valueOf(command.toString());
+                } else if (command instanceof DateTimeType) {
+                    dateTime = (DateTimeType) command;
+                }
+                if (dateTime != null) {
+                    return new SetClockCommand(dateTime.getCalendar(), bridgeHandler.getUserCode());
+                }
+                break;
+            default:
+                // do nothing for other types of status
+                break;
+        }
+
+        return null;
+    }
+
+    @Override
+    protected Collection<SatelCommand> getRefreshCommands(NewStatesEvent event) {
+        Collection<SatelCommand> result = new LinkedList<>();
+        boolean anyStatusChannelLinked = getThing().getChannels().stream()
+                .filter(channel -> STATUS_CHANNELS.contains(channel.getUID().getId()))
+                .anyMatch(channel -> !linkRegistry.getLinks(channel.getUID()).isEmpty());
```

Comment:
That's not true. `linkRegistry` is a protected field and that means it is supposed to be used by subclasses of `BaseThingHandler`. I would use `isLinked` but it lacks of version with channel object as a parameter and since I already have a channel this is a kind of optimization.

Reply:
See https://github.com/eclipse/smarthome/pull/5161

Reply:
> I would use isLinked but it lacks of version with channel object as a parameter

Well, you can easily take the id from your channel object and use this method with it.

Reply:
You're referencing something that is **hours** old in a PR 9 moths old. That's not fair...

Reply:
Haha, indeed :-)
I didn't blame you for not knowing that it wasn't supposed to be used. I only created the linked PR in order to be able to blame people in future if they use it 😎 .

Reply:
To be honest most parts of the code is over 2 years old and I am not talking about files ported from OH 1.x binding. Since API was changing frequently that time I decided to hold on with publishing 2.x version. 
It seems it still applies to the current OH2 status. In such case I will be chasing the current API in this PR and it will be never merged. I am just wondering how it goes with other bindings that has been merged recently.

Reply:
The API is pretty stable and we try our very best not to break anything. FTR, `linkRegistry` was NEVER to be used by anyone and I had an eye on this on every review. it is only now that I took action and put a deprecated marker on it, so that it is clear to everyone in the first place that it must not be used.

Reply:
Done.